### PR TITLE
Add Cuenote FC Log Import WF

### PR DESCRIPTION
**Pull request by hjmsano**
Cuenote FC is a mail delivery tool provided by YMIRLINK.
Treasure Data is able to export results from TD to Cuenote but there is no connector to get delivery logs from Cuenote to TD.
A purpose of this WF is to get the delivery settings and delivery logs from Cuenote. This helps marketers to evaluate the detailed metrics for email delivery and interactions.

### CODE REVIEW

**Comment by hjmsano** [comment]
ready to merge

File: integration-box/cuenote-import/cuenote.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        log_clickcount.set('delivid', params['delivid'])
+        log_clickcount.set('strcode', params['strcode'])
+        log_clickcount.set('with_usercolumn', '1')
+        log_clickcount.set('with_insert', '1')
+    else:
+        for key in params:
+            parameter.set(key, params[key])
+
+    req_body = "CCC=%E6%84%9B&xml=" + urllib.parse.quote(xEt.tostring(xml))
+    basic_user_and_pasword = base64.b64encode('{}:{}'.format(CNFC_USER, CNFC_PASSWORD).encode('utf-8'))
```
Comment (chezou) [inline integration-box/cuenote-import/cuenote.py]: JFYI, you can use f-string like:

```suggestion
    basic_user_and_pasword = base64.b64encode(f'{CNFC_USER}:{CNFC_PASSWORD}'.encode('utf-8'))
```

File: integration-box/cuenote-import/cuenote.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    })
+
+    with urllib.request.urlopen(request) as res:
+        data = res.read()
+        root = xEt.fromstring(data)
+        return root[0]
+
+
+def download_log(url):
+    basic_user_and_pasword = base64.b64encode('{}:{}'.format(CNFC_USER, CNFC_PASSWORD).encode('utf-8'))
```
Comment (chezou) [inline integration-box/cuenote-import/cuenote.py]: ditto

```suggestion
    basic_user_and_pasword = base64.b64encode(f'{CNFC_USER}:{CNFC_PASSWORD}'.encode('utf-8'))
```

File: integration-box/cuenote-import/cuenote_import_delivery_logs.dig [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        +count_records_in_stg:
+            td>: queries/count_records_in_stg.sql
+            store_last_results: true
+        +records_available:
+            if>: ${td.last_results['status']}
+            _do:
+                +accumulate_to_dest:
+                    td>: queries/accumulate_to_dest.sql
+
+
```
Comment (chezou) [inline integration-box/cuenote-import/cuenote_import_delivery_logs.dig]: [nits] needless line breaks

File: integration-box/cuenote-import/cuenote_import_master.dig [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        image: "digdag/digdag-python:3.7"
+    py>: get_jobs.main
+    _env:
+        td_apikey: ${secret:td.apikey}
+        td_endpoint: ${secret:td.apiserver}
+        td_database: ${secret:td.database}
+        cnfc_endpoint: ${secret:cn.endpoint}
+        cnfc_user: ${secret:cn.user}
+        cnfc_password: ${secret:cn.password}
+
```
Comment (chezou) [inline integration-box/cuenote-import/cuenote_import_master.dig]: unnecessary line breaks

File: integration-box/cuenote-import/get_jobs.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            for expid in cuenote.call_api('startExport', {'delivid': jobinfo['delivid'][i], 'strcode': 'utf8'}).iter(
+                    'expid'):
+                expids['expid'] += [int(expid.text)]
+    df_expids = pandas.DataFrame(expids.values(), index=expids.keys()).T
+
+    # Refresh Job Info table.
+    client.load_table_from_dataframe(df_jobinfo, 'jobinfo', writer='bulk_import', if_exists='overwrite')
+
+    # Insert expids into the queue table.
+    if len(expids['expid']) > 0:
```
Comment (chezou) [inline integration-box/cuenote-import/get_jobs.py]: [nits] seems unnecessary `len()`. 

```py
In [1]: bool([])
Out[1]: False
```

You can write as the following:

```suggestion
    if expids['expid']:
```

File: integration-box/cuenote-import/get_logs.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+TD_API_SERVER = os.environ.get('td_endpoint')
+TD_DATABASE = os.environ.get('td_database')
+
+
+def main():
+    # Create a TD client instance.
+    client = pytd.Client(apikey=TD_API_KEY, endpoint=TD_API_SERVER, database=TD_DATABASE)
+
+    # Download log files from Cuenote, then upload CSVs to TD
+    expids = client.query("SELECT expid FROM queue")
```
Comment (chezou) [inline integration-box/cuenote-import/get_logs.py]: If the data could be huge, you might be able to use `cursor.fetchone()` with python-presto-client https://pytd-doc.readthedocs.io/en/latest/usage.html#working-with-db-api, which loads data in a streaming manner.

File: integration-box/cuenote-import/queries/accumulate_to_dest.sql [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+       *
+    FROM
+      ${table}_stg
+;
+
+-- Drop the staging table.
+DROP
+  TABLE
+    IF EXISTS ${table}_stg
+;
```
Comment (chezou) [inline integration-box/cuenote-import/queries/accumulate_to_dest.sql]: add line break?

File: integration-box/cuenote-import/get_logs.py (hunk 6 above)
Comment (hjmsano) [reply integration-box/cuenote-import/get_logs.py]: This "queue" is a waiting list for report generation by Cuenote, and expid is created per a delivery setting for the last 14 days, so # of expid won't be huge. (is not # of recipient)

**Pull request merged**